### ISSUE TITLE
cat_breadcrumb_record / cat_breadcrumb_replay & cat_breadcrumb_autoreplay

### ISSUE THREAD

**Issue by SylveonBottle** [nullifiedcat/cathook#214]
Based off of #213

User types `cat_breadcrumb_record {something here, preferably map name and spawn point}` to start recording breadcrumbs on next spawn. Then let them walk down to the end of the map using whatever path they feel like. Do this all preferably on an offline server.

When they join the server, user would type `cat_breadcrumb_replay {the thing you saved last time}` (or `cat_breadcrumb_autoreplay {possibly automate this to auto-select based on current map/position?}` to automate this every respawn) to automatically start walking out of spawn and down the breadcrumb path previously recorded. When the user moves normally, break them out of the auto-walk path thingy.

This is some super advanced AI bullshit that automates the entire process of getting to the end of the map. If you add this with the todo autozoom, you basically have an AI that walks to the enemy spawn and spawncamps them, which seems fucking amazing

Also: This is 100% completely useless and I'll accept the "just walk yourself" response. Just putting this here in case it gets added to the TODO further down the road. The 20 second respawn time in TF2 is pretty BS, so if you automate the task of walking to wherever it would give you some more time to relax a little, maybe go take a piss.

Also, insteaad of just auto-recording your last position and having the user create their own recordings, this could let users record themselves going into a nice hidden location and replay it. Maybe the breadcrumbs thing will get smarter in the future, and we can have people recording rocket jumps to get to insane heights or places engineer can get up to, and then upload it online to a website/discord and let users simply replay the recording and let cathook automate the act of getting up to where the original recorder got to.

TL;DR: Let user start recording their own breadcrumbs and replay it back in the future.

**Comment by julianacat** [comment]
This would be a nice feature but I wouldn't know how to save this information. Even then its still useless...

**Comment by nullifiedcat** [comment]
You can save it in binary form like I do with skinchanger.

**Comment by nullifiedcat** [comment]
Implemented as walkbot.
